e2e/fuzz/runner.go

```diff
@@ -104,29 +108,52 @@ func validateNodes(c *e2e.Cluster) {
 	}
 }
 
-// validateCluster checks wheter there is enough running nodes that can make consensus
+// validateCluster checks if there is enough running nodes that can make consensus
 func validateCluster(c *e2e.Cluster) ([]string, bool) {
-	totalNodesCount := 0
+	totalNodesCount := len(c.Nodes())
 	var runningNodes []string
-	for _, n := range c.Nodes() {
-		if n.IsRunning() {
+	var partitions map[string][]string
+	// running nodes in majority partition
+	hook := c.GetTransportHook()
+	if hook != nil {
+		partitions = hook.GetPartitions()
```

Comment:
If we have a new attack like the flow map used for the liveness issue (though not sure if we will use that), we cannot have anymore this type of check. I think it does work now but we have to figure out how to scale this for any arbitrary type of partition.

Reply:
Yes, we would need to design the flow map and then to find better way of checking if there is enough nodes connected. Current implementation relays on the fact that all nodes inside partition are connected.